lib/handler/decorator-handler.js

```diff
@@ -1,35 +1,44 @@
 'use strict'
 
 module.exports = class DecoratorHandler {
+  #handler
+
   constructor (handler) {
-    this.handler = handler
+    if (typeof handler !== 'object' || handler === null) {
+      throw new TypeError('handler must be an object')
+    }
+    this.#handler = handler
   }
 
   onConnect (...args) {
-    return this.handler.onConnect(...args)
+    return this.#handler.onConnect(...args)
   }
 
   onError (...args) {
-    return this.handler.onError(...args)
+    return this.#handler.onError(...args)
   }
 
   onUpgrade (...args) {
-    return this.handler.onUpgrade(...args)
+    return this.#handler.onUpgrade(...args)
+  }
+
+  onResponseStarted (...args) {
+    return this.#handler.onResponseStarted?.(...args)
   }
 
   onHeaders (...args) {
-    return this.handler.onHeaders(...args)
+    return this.#handler.onHeaders(...args)
   }
 
   onData (...args) {
-    return this.handler.onData(...args)
+    return this.#handler.onData(...args)
   }
 
   onComplete (...args) {
-    return this.handler.onComplete(...args)
+    return this.#handler.onComplete(...args)
   }
 
   onBodySent (...args) {
-    return this.handler.onBodySent(...args)
+    return this.#handler.onBodySent(...args)
```

Comment:
```suggestion
    return this.#handler.onBodySent?.(...args)
```

I would also automatically skip un-implemented hooks.